plugins/API/Glossary.php

```diff
@@ -0,0 +1,102 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+
+namespace Piwik\Plugins\API;
+
+class Glossary
+{
+    function __construct($idSite)
+    {
+        $this->metadata = API::getInstance()->getReportMetadata($idSite);
```

Comment:
I guess ideally we would use `Report::getAllReports()`. Currently some events may add information in `getReportMetadata` but this shouldn't be the case anymore and we will refactor it in Piwik 3.0 so all information will be obtained via `getAllReports()` and `API::getReportMetadata` will be only responsible for formatting.

If that method doesn't work we should execute the API method via `Piwik\API\Request::processRequest()` and not call API methods directly see https://github.com/piwik/piwik/issues/7105